src/compiler/checker.ts

```diff
@@ -21564,6 +21563,7 @@ namespace ts {
 
         function registerForUnusedIdentifiersCheck(node: Node) {
             if (deferredUnusedIdentifierNodes) {
+                Debug.assert(!contains(deferredUnusedIdentifierNodes, node), "Registering unused identifier twice");

```

Comment:
actually, isn't the contains call slow? I imagine the list gets long.

Reply:
Yeah `contains` is a linear search - `binarySearch` would be a bit better, but best would just be using a map instead.